railties/lib/rails/commands/dbconsole/dbconsole_command.rb

```diff
@@ -51,7 +53,7 @@ def environment
     private
       def adapter_class
         ActiveRecord::ConnectionAdapters.resolve(db_config.adapter)
-      rescue LoadError, ActiveRecord::AdapterNotFound
```

Comment:
This was swallowing the helpful `AdapterNotFound` error message and raising a subsequent `NotImplementedError` on the `AbstractAdapter` it was falling back to.